akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/LogCapturing.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.testkit.typed.javadsl
+
+import scala.util.control.NonFatal
+
+import akka.actor.testkit.typed.internal.CapturingAppender
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import org.slf4j.LoggerFactory
+
+/**
+ * JUnit `TestRule` to make log lines appear only when the test failed.
+ *
+ * Use this in test by adding a public field annotated with `@TestRule`:
+ * {{{
+ *   @Rule public final LogCapturing logCapturing = new LogCapturing();
+ * }}}
+ *
+ * Requires Logback and configuration like the following the logback-test.xml:
+ *
+ * {{{
+ *     <appender name="CapturingAppender" class="akka.actor.testkit.typed.internal.CapturingAppender" />
+ *
+ *     <logger name="akka.actor.testkit.typed.internal.CapturingAppenderDelegate" >
+ *       <appender-ref ref="STDOUT"/>
+ *     </logger>
+ *
+ *     <root level="DEBUG">
+ *         <appender-ref ref="CapturingAppender"/>
+ *     </root>
+ * }}}
+ */
+final class LogCapturing extends TestRule {
+  // eager access of CapturingAppender to fail fast if misconfigured
+  private val capturingAppender = CapturingAppender.get("")
```

Comment:
`""` is translated to the root appender name.

Would be possible to install it on another logger name, but I didn't find the reason for exposing that option. Similar in LoggingEventFilter, it can be enabled for a specific logger but by default root.